AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -98,6 +100,15 @@ extension QueryPredicate {
     }
 }
 
+extension QueryPredicateConstant: GraphQLFilterConvertible {
+    func graphQLFilter(for modelSchema: ModelSchema?) -> GraphQLFilter {
+        if (self == .all) {
+            return [:]
```

Comment:
have you made sure this works at the integration level? a request that makes it to AppSync service with the GraphQLRequest, 
document: 
```
query SyncMockSynceds($filter: ModelMockSyncedFilterInput, $limit: Int) {
              syncMockSynceds(filter: $filter, limit: $limit) {
                items {
                  id
                  __typename
                  _version
                  _deleted
                  _lastChangedAt
                }
                nextToken
                startedAt
              }
            }
```

what actually ends up being in the `variables` object?

Reply:
when i try on the AWS Console on a simplifed example

### docment with filter parameter, and no "filter" variables
document
```
query MyQuery($filter: ModelCommentFilterInput) {
  listComments(filter: $filter, limit: 10) {
    items {
      content
    }
  }
}

```
variables is empty

```
```

This works and response is successful

### document with filter parameter, and "filter" variables
```
query MyQuery($filter: ModelCommentFilterInput) {
  listComments(filter: $filter, limit: 10) {
    items {
      content
    }
  }
}

```
variables is empty object: `{}`

```
{
  "filter": {}
}
```

response:
```
{
  "data": {
    "listComments": null
  },
  "errors": [
    {
      "path": [
        "listComments"
      ],
      "data": null,
      "errorType": "DynamoDB:DynamoDbException",
      "errorInfo": null,
      "locations": [
        {
          "line": 2,
          "column": 3,
          "sourceName": null
        }
      ],
      "message": "ExpressionAttributeNames must not be empty (Service: DynamoDb, Status Code: 400, Request ID: 8P5G95PBK3R7GTA1MLHJ5VKFMNVV4KQNSO5AEMVJF66Q9ASUAAJG, Extended Request ID: null)"
    }
  ]
}
```

Reply:
So i'm guessing the filter object `[:]` gets added to variables, which gets added to the request payload as an object that looks like this:
```
let queryDocument = 
[ 
 "document": "some graphql document that references filter input",
 "variables": [
    "filter": [:]
  ]
]
```
and [serialized here](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLOperation.swift#L81) using JSONSerialization, that probably looks like this as the JSON string?
```
{ 
 "document": "some graphql document that references filter input",
 "variables": {
    "filter": {}
  }
}
```
is this what's happening? can we verify this use case is working in an integration test?